vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthEmailEntryFragment.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.onboarding.ftueauth
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.lifecycle.lifecycleScope
+import im.vector.app.core.extensions.associateContentStateWith
+import im.vector.app.core.extensions.content
+import im.vector.app.core.extensions.editText
+import im.vector.app.core.extensions.isEmail
+import im.vector.app.core.extensions.setOnImeDone
+import im.vector.app.databinding.FragmentFtueEmailInputBinding
+import im.vector.app.features.onboarding.OnboardingAction
+import im.vector.app.features.onboarding.OnboardingViewState
+import im.vector.app.features.onboarding.RegisterAction
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import org.matrix.android.sdk.api.auth.registration.RegisterThreePid
+import reactivecircus.flowbinding.android.widget.textChanges
+import javax.inject.Inject
+
+class FtueAuthEmailEntryFragment @Inject constructor() : AbstractFtueAuthFragment<FragmentFtueEmailInputBinding>() {
+
+    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentFtueEmailInputBinding {
+        return FragmentFtueEmailInputBinding.inflate(inflater, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupViews()
+    }
+
+    private fun setupViews() {
+        views.emailEntryInput.associateContentStateWith(button = views.emailEntrySubmit)
+        views.emailEntryInput.setOnImeDone { updateEmail() }
+        views.emailEntrySubmit.debouncedClicks { updateEmail() }
+
+        views.emailEntryInput.editText().textChanges()
+                .onEach {
+                    views.emailEntryInput.error = null
+                    views.emailEntrySubmit.isEnabled = it.isEmail()
+                }
+                .launchIn(lifecycleScope)
```

Comment:
It's safer to use `viewLifecycleOwner.lifecycleScope`

Reply:
great catch, will do :+1:

Reply:
https://github.com/vector-im/element-android/pull/5868/commits/b2e3dd15a4c22784b2c25d7a0a1ca25fb7cbc8a4 updated other usages as well